tests/test_testclient.py

```diff
@@ -14,15 +25,54 @@ def mock_service_endpoint(request):
     return JSONResponse({"mock": "example"})
 
 
-def create_app(test_client_factory):
-    app = Starlette()
+_identity_runvar: anyio.lowlevel.RunVar[int] = anyio.lowlevel.RunVar("_identity_runvar")
+
+
+def get_identity(counter):
+    try:
+        return _identity_runvar.get()
+    except LookupError:
+        token = next(counter)
+        _identity_runvar.set(token)
+        return token
+
+
+def current_task():
+    # anyio's TaskInfo comparisons are invalid after their associated native
+    # task object is GC'd https://github.com/agronholm/anyio/issues/324
+    asynclib_name = sniffio.current_async_library()
+    if asynclib_name == "trio":
+        return trio.lowlevel.current_task()
+
+    if asynclib_name == "asyncio":
+        task = asyncio_current_task()
+        if task is None:
+            raise RuntimeError("must be called from a running task")  # pragma: no cover
+        return task
+    raise RuntimeError(f"unsupported asynclib={asynclib_name}")  # pragma: no cover
+
+
+def create_app(test_client_factory, counter=itertools.count()):
```

Comment:
Think this `counter=itertools.count()` is the classic state-in-a-param surprising Python thing, is that intentional?
```suggestion
def create_app(test_client_factory, counter=None):
    if counter is None:
        counter = itertools.count()
```

Reply:
Yes it's intentional so the thread count increases globally unless a counter is passed in